sunxspex/thermal.py

```diff
@@ -0,0 +1,699 @@
+import copy
+
+import astropy.units as u
+import numpy as np
+from scipy import interpolate, stats
+from sunpy.data import manager
+
+from sunxspex.io import load_chianti_continuum, load_chianti_lines_lite, load_xray_abundances
+
+doc_string_params = """
+Parameters
+----------
+energy_edges: `astropy.units.Quantity`
+    The edges of the energy bins in a 1D N+1 quantity.
+
+temperature: `astropy.units.Quantity`
+    The temperature of the plasma.
+    Can be scalar or 1D of any length. If not scalar, the flux for each temperature
+    will be calculated. The first dimension of the output flux will correspond
+    to temperature.
```

Comment:
```suggestion
    Quantity can be scalar or 1D of any length.
    If not scalar, the flux at each temperature will be calculated separately and
    will be accessed via the first axis of the returned flux array.
    If scalar, the returned flux array will not have a temperature axis.
```

Does this make more sense @hayesla?

Reply:
I was trying to preserve the dimensionality of the inputs in the same way slicing does.  It felt more intuitive, although I grant always having a temperature dimension is easier to describe in a docstring.